File: News_Service_Jooq/src/main/java/com/dunice/projectNews/services/UserService.java
package com.dunice.projectNews.services;

import com.dunice.projectNews.dto.request.PutUserRequest;
import com.dunice.projectNews.dto.response.PublicUserResponse;
import com.dunice.projectNews.models.News;
import com.dunice.projectNews.models.User;

import java.util.List;
import java.util.UUID;

public interface UserService {
    List<PublicUserResponse> getAllUser();

    PublicUserResponse replaceUser(PutUserRequest publicUserView);

    void deleteUser();

    PublicUserResponse getUserInfoById(UUID id);

    PublicUserResponse getUserInfo();

    void saveUserForNews(News news);

    Boolean checkingUserCurrentAuthenticationUser(User user);
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/services/UserService.kt
package com.dunice.advancedServiceKotlin.services

import News
import com.dunice.advancedServiceKotlin.dto.request.PutUserRequest
import com.dunice.advancedServiceKotlin.dto.response.PublicUserResponse
import com.dunice.advancedServiceKotlin.models.User
import java.util.*

interface UserService {

    val allUser: List<PublicUserResponse?>?

    fun replaceUser(publicUserView: PutUserRequest?): PublicUserResponse?

    fun deleteUser()

    fun getUserInfoById(id: UUID): PublicUserResponse?

    val userInfo: PublicUserResponse?

    fun saveUserForNews(news: News)

    fun checkingUserCurrentAuthenticationUser(user: User): Boolean
}
